Main/Main.ino

```diff
@@ -0,0 +1,149 @@
+                    GCLK_GENDIV_ID(4);            // Select Generic Clock (GCLK) 4
+  while (GCLK->STATUS.bit.SYNCBUSY);              // Wait for synchronization
+
+  REG_GCLK_GENCTRL = GCLK_GENCTRL_IDC |           // Set the duty cycle to 50/50 HIGH/LOW
+                     GCLK_GENCTRL_GENEN |         // Enable GCLK4
+                     GCLK_GENCTRL_SRC_DFLL48M |   // Set the 48MHz clock source
+                     GCLK_GENCTRL_ID(4);          // Select GCLK4
+  while (GCLK->STATUS.bit.SYNCBUSY);              // Wait for synchronization
+
+  Serial.println("GLK was setup");
+
+  // Feed GCLK4 to TC4 and TC5
+  REG_GCLK_CLKCTRL = GCLK_CLKCTRL_CLKEN |         // Enable GCLK4 to TC4 and TC5
+                     GCLK_CLKCTRL_GEN_GCLK4 |     // Select GCLK4
+                     GCLK_CLKCTRL_ID_TC4_TC5;     // Feed the GCLK4 to TC4 and TC5
+  while (GCLK->STATUS.bit.SYNCBUSY);              // Wait for synchronization
+ 
+  REG_TC4_CTRLA |= TC_CTRLA_MODE_COUNT32;           // Set the counter to 32-bit mode - This will use both TC4 & TC5 to count 
+  while (TC4->COUNT32.STATUS.bit.SYNCBUSY);        // Wait for synchronization
+
+
+  REG_TC4_COUNT32_CC0 = 0x7270E;                  // Set the counter top register to trigger after 30 sec, 15625Hz (below) * 30 = 468750 = 0x7270E
+  while (TC4->COUNT32.STATUS.bit.SYNCBUSY);        // Wait for synchronization
+  
+
+  Serial.println("Timer set-up complete. Enabling interrupt"); // Debug
+  
+  NVIC_SetPriority(TC4_IRQn, 0);    // Set the Nested Vector Interrupt Controller (NVIC) priority for TC4 to 0 (highest)
+  NVIC_EnableIRQ(TC4_IRQn);         // Connect TC4 to Nested Vector Interrupt Controller (NVIC)
+
+  REG_TC4_INTFLAG |= TC_INTFLAG_MC1 | TC_INTFLAG_MC0 | TC_INTFLAG_OVF;        // Clear the interrupt flags to be safe
+  REG_TC4_INTENCLR = TC_INTENCLR_MC0;     // Disable TC4 CC0 interrupt
+
+ 
+  REG_TC4_CTRLA |= TC_CTRLA_PRESCALER_DIV1024 |     // Set prescaler to 1024, 48MHz/3(GCLK divider)/1024(prescaler) = 15625 Hz
+                   TC_CTRLA_ENABLE;               // Enable TC4
+  while (TC4->COUNT32.STATUS.bit.SYNCBUSY);        // Wait for synchronization
+  
+  Serial.println("Timer interrupt set-up complete");
+}
+
+
+
+
+
+
+void setup() {
+  //Assign LED pin and PIR pin
+  pinMode(LED, OUTPUT);
+  pinMode(PIR, INPUT);
+
+  Serial.begin(9600);
+
+  
+
+  Serial.println("Running Timer set-up");
+  timer_setup();
+
+  // Indicate start-up of device with LED.
+  digitalWrite(LED, HIGH);
+  delay(2000);
+  digitalWrite(LED, LOW);
+
+  // Assign vehicle detection interrupt to the PIR pin, when pin pulled LOW
+  attachInterrupt(digitalPinToInterrupt(PIR), vehicle_detection_ISR, LOW);
+  
+}
+
+
+void loop() {
+  
+  // If the PIR has been triggered and flag is raised then flash the LED
+  while (PIR_Detection_Flag == 1){
+    digitalWrite(LED, HIGH);
+    delay(500);
+    digitalWrite(LED, LOW);
+    delay(500);
```

Comment:
In future we should move this to a separate timer since there will be WiFi code happening in the background. The delay() might mess with it. This can be done in future